### PR TITLE
Various refactoring

### PR DESCRIPTION
**Pull request by boxdot**

### CODE REVIEW

File: lib/intersection.h [hunk 1]
```diff
@@ -61,8 +60,7 @@ inline float intersect_ray_plane(const Ray& ray, const Vec& v0, const Vec& n) {
  *   true, if the ray intersects the triangle, otherwise false
  */
 inline bool intersect_ray_triangle(const Ray& ray, const Triangle& tri,
```
Comment (jeschkies) [inline lib/intersection.h]: How wide is this line? I'm looking at the diff on my 13'' Mac with split view. Ideally the view does not break lines. That's an argument for 80 character width.

File: lib/solid_angle.h [hunk 2]
```diff
@@ -1,111 +0,0 @@
...
- * Compute the solid angle subtended by triangle from `O`.
- *
- * See Eriksson, Folke (1990). "On the measure of solid angles". Math. Mag. 63
- * (3): 184–187.
- *
- * @param  O       origin
- * @param  A, B, C vertices of the triangle
- * @return         solid angle
- */
-float solid_angle(const Point& O, const Point& A, const Point& B,
```
Comment (jeschkies) [inline lib/solid_angle.h]: Where did this go?

File: lib/intersection.h (hunk 1 above)
Comment (boxdot) [inline lib/intersection.h]: I am counting in my editor 71 chars.

File: lib/solid_angle.h (hunk 2 above)
Comment (boxdot) [inline lib/solid_angle.h]: > Merge radiosity functions and tests.  …
> * Move solid angle functions and test to lib/radiosity.h.
> * Move implementations of form factor functions to header, s.t.
>   they can be inlined.

We are basically using this function only in radiosity, so moved them into the library file of radiosity.

**Comment by boxdot** [comment]
> How did you do the formatting? We should document that.
> Also, ideally we don't have line breaks in a split view on a 13'' Macbbook.

Formatting was done with the standard tool:
```
clang-format -i *.cpp *.h lib/*.h lib/*.cpp tests/*.cpp
```
Its runtime arguments are documented. The interesting stuff is in `.clang-format`. I've added a comment there how to run the formatter. The column width is at 80. So it should be readable without any problems on your mac (with reasonable font size ;)).

**Pull request merged**